apps/graylog/restapi/custom/api.pm

```diff
@@ -0,0 +1,247 @@
+
+    return $self;
+}
+
+sub set_options {
+    my ($self, %options) = @_;
+
+    $self->{option_results} = $options{option_results};
+}
+
+sub set_defaults {}
+
+sub check_options {
+    my ($self, %options) = @_;
+
+    $self->{hostname} = (defined($self->{option_results}->{hostname})) ? $self->{option_results}->{hostname} : undef;
+    $self->{port} = (defined($self->{option_results}->{port})) ? $self->{option_results}->{port} : 9000;
+    $self->{proto} = (defined($self->{option_results}->{proto})) ? $self->{option_results}->{proto} : 'http';
+    $self->{url_path} = (defined($self->{option_results}->{url_path})) ? $self->{option_results}->{url_path} : '/api/';
+    $self->{timeout} = (defined($self->{option_results}->{timeout})) ? $self->{option_results}->{timeout} : 10;
+    $self->{credentials} = (defined($self->{option_results}->{credentials})) ? 1 : undef;
+    $self->{basic} = (defined($self->{option_results}->{basic})) ? 1 : undef;
+    $self->{timeframe} = (defined($self->{option_results}->{timeframe})) ? $self->{option_results}->{timeframe} : 300;
+    $self->{step} = (defined($self->{option_results}->{step})) ? $self->{option_results}->{step} : undef;
+ 
+    if (!defined($self->{hostname}) || $self->{hostname} eq '') {
+        $self->{output}->add_option_msg(short_msg => "Need to specify hostname option.");
+        $self->{output}->option_exit();
+    }
+    
+    return 0;
+}
+
+sub build_options_for_httplib {
+    my ($self, %options) = @_;
+
+    $self->{option_results}->{hostname} = $self->{hostname};
+    $self->{option_results}->{timeout} = $self->{timeout};
+    $self->{option_results}->{port} = $self->{port};
+    $self->{option_results}->{proto} = $self->{proto};
+    $self->{option_results}->{url_path} = $self->{url_path};
+    $self->{option_results}->{credentials} = $self->{credentials};
+    $self->{option_results}->{basic} = $self->{basic};
+    $self->{option_results}->{warning_status} = '';
+    $self->{option_results}->{critical_status} = '';
+}
+
+sub settings {
+    my ($self, %options) = @_;
+
+    $self->build_options_for_httplib();
+    $self->{http}->set_options(%{$self->{option_results}});
+}
+
+sub get_connection_info {
+    my ($self, %options) = @_;
+    
+    return $self->{hostname} . ":" . $self->{port};
+}
+
+sub get_hostname {
+    my ($self, %options) = @_;
+    
+    return $self->{hostname};
+}
+
+sub get_port {
+    my ($self, %options) = @_;
+    
+    return $self->{port};
+}
+
+sub query_absolute {
+    my ($self, %options) = @_;
+
+    my $start_time = DateTime->now->subtract(seconds => $options{timeframe}).'.000';
+    $start_time =~ s/T/ /;
+    $start_time =~ s/ /%20/;
+    my $end_time = DateTime->now.'.000';
+    $end_time =~ s/T/ /;
+    $end_time =~ s/ /%20/;
+    my $uri = URI::Encode->new({encode_reserved => 1});
+
+    my $result = $self->get_endpoint(url_path => 'search/universal/absolute?query=' . $uri->encode($options{query}) .
```

Comment:
Put it in 'get_param' attribute (and it will encoded by the library)

Be sure to get that format: yyyy-MM-dd HH:mm:ss (in UTC)
But i think that endpoint it's easier: '/search/universal/relative' . You can use 'range' get attribute (in a relative timerange, specified as seconds from now. )